tests/PSRule.Rules.Azure.Tests/Azure.AKS.Tests.ps1

```diff
@@ -293,6 +293,22 @@ Describe 'Azure.AKS' -Tag AKS {
             $ruleResult.Length | Should -Be 1;
             $ruleResult.TargetName | Should -BeIn 'cluster-F';
         }
+
+        It 'Azure.AKS.AzureCNI' {
+            $filteredResult = $result | Where-Object { $_.RuleName -eq 'Azure.AKS.AzureCNI' };
+
+            # Fail
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Fail' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult | Should -HaveCount 3;
+            $ruleResult.TargetName | Should -BeIn 'cluster-B', 'cluster-D', 'system';
+
+            # Pass
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Pass' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult | Should -HaveCount 3;
+            $ruleResult.TargetName | Should -BeIn 'cluster-A', 'cluster-C', 'cluster-F';
```

Comment:
You can probably include one example without any sub-resource to test that scenario. Maybe remove the sub-resource from cluster-A. Since the other test resources have the sub-resource.

Reply:
Remove sub resources from cluster-A. That scenario should be covered now.